### ISSUE TITLE
Don't save sprite property for badguys if the default sprite is used

### ISSUE THREAD

**Issue by tobbi** [SuperTux/supertux#781]
I see a lot of these things in levels created by the editor:
```
    (bouncingsnowball
      (sprite "images/creatures/bouncing_snowball/bouncing_snowball.sprite")
      (x 433.1000061)
      (y 1029.099976)
      (direction "left")
    )
```

As you can see, we're saving the sprite property with it, even when the default sprite is used.
We shouldn't do that, it makes our level files larger than they should be.

Since I'll be working on the editor menu when I get the chance, this will be up for grabs by the rest of the community.

**Comment by durgeshra** [comment]
I would like to give it a shot. Can you elaborate a bit on what I am supposed to do in this issue?

**Comment by tobbi** [comment]
I was thinking about a virtual ``get_default_sprite`` method in the ``Badguy`` class that returns  the path to the default sprite and gets overridden by every child class.

Then we could compare with that in the method that saves the sprite file path and only save it if it's different from the default.

In the process, you will probably also have to change the constructors of ``Badguy`` (and their calls) to not include the sprite file path anymore, but get it from the ``get_default_sprite`` method.

I hope I'm making sense here. If not, feel free to ask.

**Comment by durgeshra** [comment]
I am a newbie and it did not get into my mind very much how I should proceed. Which file am I supposed to change or am I supposed to create a new file altogether? And which language are we using here?

**Comment by tobbi** [comment]
We're using C++. How good are you at C++? The concepts I'm describing are somewhat advanced.

**Comment by maxteufel** [comment]
Just a side note, this problem isn't only affecting sprites. It's affecting everything that has a default value. It gets stored by the editor in the level files. Default properties shouldn't and don't need to be saved. Since default values are already stored somewhere in the class definition, I think adding a separate function is a silly thing to do. We should take the information from where it already is.

**Comment by maxteufel** [comment]
@durgeshra Well, we're trying to solve the problem as stated in the first post, and more generally also the problem I talked about in my last comment. If the problem as described in the first post does not say anything to you, it might be a good step to familiarize yourself with the way SuperTux works and the exact problem (this issue more specifically targets the way levels are stored by the editor). Thus, look into the level format, and then the editor first.

I have to admit, this seems to be a pretty hard problem for someone who does not know anything about the SuperTux codebase. I would look into more smaller problems that help you dive into the codebase, such as those listed with the following labels:

- [difficulty:easy](https://github.com/SuperTux/supertux/issues?q=is%3Aopen+is%3Aissue+label%3Adifficulty%3Aeasy)
- [good first issue](https://github.com/SuperTux/supertux/issues?q=is%3Aopen+is%3Aissue+label%3A%22good+first+issue%22)
- [help wanted](https://github.com/SuperTux/supertux/issues?q=is%3Aopen+is%3Aissue+label%3A%22help+wanted%22)

If you have any (specific) questions, it would be useful to either ask them in the issue's comments, or by getting in touch with us via [IRC](https://kiwiirc.com/client/chat.freenode.net:+6697/?nick=Guest?#supertux) or [Matrix](https://riot.im/app/#/room/#supertux:matrix.org) (note that on these real-time chat platforms, we might not be online all day, so please *wait* for answers).

**Comment by J-128** [comment]
@Grumbel does anyone actually _use_ the C# editor anymore?  I know I don't. I use the in-game editor. :thinking:

**Comment by Grumbel** [comment]
Fixed with 3dda16f42
